src/ninja.cc

```diff
@@ -955,8 +1021,8 @@ int NinjaMain::ToolCompilationDatabase(const Options* options, int argc,
   }
 
   putchar('[');
-  for (vector<Edge*>::iterator e = state_.edges_.begin();
-       e != state_.edges_.end(); ++e) {
+  for (vector<Edge*>::iterator e = edges_to_process.begin();
+       e != edges_to_process.end(); ++e) {
```

Comment:
since we're C++11 now, can we switch to proper for-loops now when modifying the code? This seems a good opportunity to do that here.